lte/gateway/c/session_manager/SetMessageManagerHandler.cpp

```diff
@@ -128,11 +173,42 @@ void SetMessageManagerHandler::send_create_session(
     } else {
       MLOG(MERROR)
           << "Failed to initialize 5G session for subscriber"
-          << cfg.common_context.sid().id() << " with PDU session ID "
+          << cfg.common_context.sid().id() << " with PDU session ID  from UE"
           << cfg.rat_specific_context.m5gsm_session_context().pdu_session_id()
           << " due to failure writing to SessionStore.";
     }
   }
 }
 
+/* This starts releasing the session in main session enforcer thread context
+ * Before startting it checks if respective session */
+void SetMessageManagerHandler::initiate_release_session(
+    SessionMap& session_map, const std::string& dnn, const std::string& imsi) {
+  // TODO as modification and dynamic rules are not implemented this may
+  // return empty map.
+  auto update = SessionStore::get_default_session_update(session_map);
+  bool exist =
+      m5g_enforcer_->m5g_release_session(session_map, imsi, dnn, update);

```

Comment:
where is the definition of this function?
Why is the session released by imsi and not by session_id?

Reply:
Release session request comes from AMF, AMF doesn't know the session_id.  On a unique IMSI there can multiple sessions cud be created, all wud have a unique "dnn".  That is reason while releasing a session we are  taking "imsi" and "dnn" as inputs.